### DISCUSSION TITLE
Is there a way to automatically stop and start the container on connection that doesn't reserve RAM?

### DISCUSSION THREAD

**Discussion by Qesaru** [itzg/docker-minecraft-server#2181] [Q&A]
Hey everyone,
I'm running this awesome container in Unraid. And I like the ease of use so much that I'll probably ditch my existing Pterodactyl Setup and use this container instead.

However I have one problem before I can do so.
I am quite limited on RAM on my system and I want to save some power. Therefore I'm a big fan of the Server Pause and Server Stop features. However both come with some difficulties on their own:
The Server Pause Feature doesn't release the RAM usage on my system. I know from another support thread that modern Linux kernels will just dump the currently unused memory to the swap file when the system needs more memory. However Unraid doesn't use a swap file by default. Therefore I'd need to find a workaround for that. I know that there is a possibility to get swap running, but I wanted to ask if there might be an easier way before I commit to using swap on a system that's not intended to do so.
The Server Stop Feature would work great with things like [MCSleepingServerStarter](https://github.com/vincss/mcsleepingserverstarter). However this software would only work within the container. Therefore I'd need a way to change the start-command to start SleepingServerStarter instead of the MC Server instance itself. Also I'd need the Container to stay active even after the Minecraft Server has automatically stopped.

So basically I'm asking for one of 3 solutions:
1. Make Auto Pause release the RAM
2. Get MCSleepingServerStarter to run inside the container on startup and force the container to stay active even after Auto Stop shut down the MC server instance
3. Find a way to start up the container when someone wants to connect to the server in a way that is easily accessible for everyone. Either via a mobile app, a discord/telegram bot or - even better - from within minecraft by just trying to connect.

What might be the best approach to do this?
Thank you in advance!

**Comment by itzg** [comment]
I think there's one or two enhancement requests to get end to end stop/start, but it's not trivial so keeps falling behind in the long TODO list.
